src/sys/select.rs

```diff
@@ -88,11 +97,13 @@ impl FdSet {
     ///
     /// ```
     /// # use nix::sys::select::FdSet;
-    /// # use std::os::unix::io::RawFd;
+    /// # use std::os::unix::io::{AsRawFd, BorrowedFd, RawFd};
     /// let mut set = FdSet::new();
-    /// set.insert(4);
-    /// set.insert(9);
-    /// let fds: Vec<RawFd> = set.fds(None).collect();
+    /// let fd_four = unsafe {BorrowedFd::borrow_raw(4)};
+    /// let fd_nine = unsafe {BorrowedFd::borrow_raw(9)};
+    /// set.insert(&fd_four);
+    /// set.insert(&fd_nine);
+    /// let fds: Vec<RawFd> = set.fds(None).map(|borrowed_fd|borrowed_fd.as_raw_fd()).collect();
     /// assert_eq!(fds, vec![4, 9]);
     /// ```
     #[inline]
```

Comment:
The `highest` parameter of this `fds()` function, IMO, using a `RawFd`  is better than using a `AsFd` as users may just wanna specify a specific value, and this value (fd) probably is not open. For example, I just wanna construct an iterator whose file decriptors are smaller than 50:

```rust
let set = FdSet::new();
// some insertions..
let fds_smaller_than_fifty = set.fds(Some(50));
```
I don't want to create a `BorrowedFd` first, which is obviously too tedious:

```rust
let fd_fifty = unsafe { BorrowedFd::borrow_raw(50)) };
let fds = set.fds(Some(&fd_fifty));
```